### ISSUE TITLE
ConvertOnDemand objects doesn't convert when using enumerateObjectsUsingBlock:

### ISSUE THREAD

**Issue by ogantopkaya** [icanzilb/JSONModel#262]
I have provide a little snippet which causes the problem.

```objc 
// In the JSONModel subclass
@property(strong,nonatomic) NSArray <Optional,ConvertOnDemand,Player> *players;
```

```objc
[jsonObject.players enumerateObjectsUsingBlock:^(Player *player, NSUInteger idx, BOOL *stop) {
      NSLog(@"%@",player.playerId); // This causes crash. Because player's class is NSDictionary
}
```

```objc
     NSLog(@"%@",jsonObject.players[0]); // This works fine.
```

**Comment by billinghamj** [comment]
The latter is Obj-C "fast enumeration", which is effectively identical to using the `enumerateObjectsUsingBlock:` method.

I would expect this to be fairly easily resolvable.

**Comment by billinghamj** [comment]
It looks like this should have been resolved by 5800a0f27b96040124e6f58b616945ea84d8f5cb. Please make a new issue if the issue persists.

**Comment by lichuanjun** [comment]
also [NSArray arrayWithArray:jsonObject.players] will crash,cuz jsonObject.players now is not a NSArray class.
